### ISSUE TITLE
Restart of gpu-operator components when a new node added

### ISSUE THREAD

**Issue by xhejtman** [NVIDIA/gpu-operator#433]
### 1. Quick Debug Checklist
- [ ] Are you running on an Ubuntu 18.04 node?
no
- [ ] Are you running Kubernetes v1.13+?
yes
- [ ] Are you running Docker (>= 18.06) or CRIO (>= 1.13+)?
no
- [ ] Do you have `i2c_core` and `ipmi_msghandler` loaded on the nodes?
yes
- [ ] Did you apply the CRD (`kubectl describe clusterpolicies --all-namespaces`)
yes

### 1. Issue or feature description

It seems that if you have existing cluster with gpu-operator installed (we have 1.11.0) with A40 cards and add a new node with A100 card, whole gpu-operator is restarted (i.e., all daemon sets are restared). This is quite unfortunate as GPUs are in use in the cluster which leads to gpu-operator fail as kernel module cannot be unloaded. 

Is this expected? We have mig strategy `single`. Does `mixed` strategy solve this issue?

**Comment by shivamerla** [comment]
@xhejtman does the new node has the same OS type/version as other nodes? Whenever a new node is added the Daemonset spec doesn't change but a new pod is spawned on the node based on the nodeSelector. Wondering what caused Daemonset spec to be changed here.

**Comment by xhejtman** [comment]
OS type is Ubuntu on all nodes, but the new node has 22.04 version, the others have 20.04, is this the reason?

**Comment by shivamerla** [comment]
Yes, that can cause it. Currently the limitation is that we support single OS version across the cluster(i.e single Daemonset with OS version image). This is noted [here](https://docs.nvidia.com/datacenter/cloud-native/gpu-operator/release-notes.html#known-limitations). There is work in progress to support heterogenous nodes.

**Comment by cdesiniotis** [comment]
@xhejtman the NVIDIADriver CRD makes it possible to have different OS versions in the same cluster: https://docs.nvidia.com/datacenter/cloud-native/gpu-operator/latest/gpu-driver-configuration.html

If this issue is still relevant with the latest version of the NVIDIA GPU Operator, please feel free to reopen it or open a new one with updated details.
